### PR TITLE
Fix error messages for some illegal constructions, fix absolute path includes

### LINKED ISSUE #242: Contract declarations under namespace
The following contract glitches out throwing `unknown_error`:
```
namespace N =                                                               
  contract C =                                                              
    entrypoint f : () => int
contract C =
  entrypoint f(c : N.C) = c.f()
```

### LINKED ISSUE #243: type declarations are not working
Type declarations are not working. The following contract fails to compile with `unknown_error` (it's a `type_error`):
```
contract N =                                  
  type d                                       
                                               
contract C =                                 
  entrypoint f() = 123
```

### LINKED ISSUE #244: Constant definitions are accepted by the parser but not by the typechecker
The following contract causes the typechecker to throw an `unknown_error`:
```
contract C =
  let x = 2138
  entrypoint f() =123
```

### LINKED ISSUE #246: Cannot include by absolute path
The following contract fails with `case_clause` (assuming file exists):
```
include "/tmp/Jebudu.aes"
contract C =
  entrypoint f() = 123
```

### LINKED ISSUE #252: Crash when printing error message
Given the contract-stub:
```
contract HashTimeLock =
  datatype status = INVALID | ACTIVE | REFUNDED | WITHDRAWN | EXPIRED
  record lock_contract = {input_amount : int, output_amount : int, expiration : int, hash_lock : hash, status: status, sender : address, receiver : address, output_network : string, output_address : string}
  entrypoint new_contract(l : lock_contract) : lock_contract = l
```

The following call-result decoding fails:
```
./aesophia_cli /tmp/foo.aes -b fate --call_result cb_q58BgUFsUQgA37pD+/b+7SyBTPjsmp5mKp47Akx0c4yLS635b4gbwW1nTsf/wG+IG8FtZ07H/8BvhgFxEO8ARp8BgYcZ2AhdUzRj15KZWrT0vIXnJEq1gEZLA1zDge3Qam24r4UAAAAAAAE/nwCgBwwURA4zMENRZdpeBEtaqcNdLL4wyRwyFFVJN19SUmSfAKBrs4jtfFRU9fsXJtVRgxnm+6MvasbSWpfv9LbvZBN72A1BRVPRYWtfNDcxZFlVclE4RUF0bXp3dUtEdzRWQkdRZG5FY1A1WUY1NjNXRzR5UjlXdmZwNXRScCV612Q= --call_result_fun new_contract 
escript: exception error: no function clause matching 
                 aeso_pretty:type({variant_t,
                                   [{constr_t,
                                     [{file,no_file},{line,2},{col,21}],
                                     {con,
                                      [{file,no_file},{line,2},{col,21}],
                                      "INVALID"},
                                     []},
                                    {constr_t,
                                     [{file,no_file},{line,2},{col,31}],
                                     {con,
                                      [{file,no_file},{line,2},{col,31}],
                                      "ACTIVE"},
                                     []},
                                    {constr_t,
                                     [{file,no_file},{line,2},{col,40}],
                                     {con,
                                      [{file,no_file},{line,2},{col,40}],
                                      "REFUNDED"},
                                     []},
                                    {constr_t,
                                     [{file,no_file},{line,2},{col,51}],
                                     {con,
                                      [{file,no_file},{line,2},{col,51}],
                                      "WITHDRAWN"},
                                     []},
                                    {constr_t,
                                     [{file,no_file},{line,2},{col,63}],
                                     {con,
                                      [{file,no_file},{line,2},{col,63}],
                                      "EXPIRED"},
                                     []}]}) (/Users/hans/Quviq/Aeternity/aesophia_cli/_checkouts/aesophia/src/aeso_pretty.erl, line 245)
  in function  aeso_pretty:typed/2 (/Users/hans/Quviq/Aeternity/aesophia_cli/_checkouts/aesophia/src/aeso_pretty.erl, line 131)
  in call from lists:map/2 (lists.erl, line 1239)
  in call from lists:map/2 (lists.erl, line 1239)
  in call from aeso_pretty:typedef/1 (/Users/hans/Quviq/Aeternity/aesophia_cli/_checkouts/aesophia/src/aeso_pretty.erl, line 228)
  in call from aeso_compiler:to_sophia_value/5 (/Users/hans/Quviq/Aeternity/aesophia_cli/_checkouts/aesophia/src/aeso_compiler.erl, line 339)
  in call from aesophia_cli:decode_call_res/5 (/Users/hans/Quviq/Aeternity/aesophia_cli/src/aesophia_cli.erl, line 269)
  in call from aesophia_cli:main/1 (/Users/hans/Quviq/Aeternity/aesophia_cli/src/aesophia_cli.erl, line 62)
```

The problem comes from trying to print an error message for a variant-type that case is missing since `aeso_pretty:type` isn't normally called directly, but in `to_sophia_value` it is.

### PR DESCRIPTION
**Pull request by radrow**
fixes #246 
fixes #244 
fixes #243 
fixes #242 
fixes #252

### CODE REVIEW

**Comment by radrow** [comment]
Also fixed attributes pretty printing

File: test/aeso_compiler_tests.erl [hunk 1]
```diff
@@ -379,7 +379,7 @@ failing_contracts() ->
         [<<?Pos(2, 11)
            "Include of 'included.aes' at line 2, column 11\nnot allowed, include only allowed at top level.">>,
          <<?Pos(3, 13)
-           "Nested namespace not allowed\nNamespace 'Foo' at line 3, column 13 not defined at top level.">>])
+           "Nested namespaces are not allowed\nNamespace 'Foo' at line 3, column 13 not defined at top level.">>])
     , ?TYPE_ERROR(bad_address_literals,
```
Comment (hanssv) [inline test/aeso_compiler_tests.erl]: There were quite a few new error messages - a test case for each would be nice :-)

Comment (radrow) [reply test/aeso_compiler_tests.erl]: Test cases are missing not only for the new messages. Imo this is a separate job to do.

Comment (hanssv) [reply test/aeso_compiler_tests.erl]: Two wrongs doesn't equal one right, or whatever the saying is :-)

Comment (radrow) [reply test/aeso_compiler_tests.erl]: If you say so. I will extend the scope of the PR and add tests for all the error messages

Comment (radrow) [reply test/aeso_compiler_tests.erl]: Done

**Comment by radrow** [comment]
Fixed type printing in `to_sophia_value` in fate errors

**Comment by radrow** [comment]
Todo: fix sophia grammar in docs

**Comment by radrow** [comment]
Okay, now it is ready (I hope)

**Pull request merged**